### PR TITLE
Integration - release 1

### PR DESCRIPTION
**Pull request by matthew-ardi**

### CODE REVIEW

File: RealtimeAppBackend/RealtimeApp/views.py [hunk 1]
```diff
@@ -43,21 +43,55 @@ def retrieveData(request):
...
+    roomLightState = event_state[0]
+    benchLightState = event_state[1]
+    blenderState = event_state[2]
+    spaceHeaterState = event_state[3]
+    roomOccupiedState = event_state[4]
+    grinderState = event_state[5]
 
     # @TODO:               then modify the global object data
-
+    data(roomLightState, benchLightState, blenderState, spaceHeaterState, roomOccupiedState, grinderState)
```
Comment (matthew-ardi) [inline RealtimeAppBackend/RealtimeApp/views.py]: @sonminhtran1997 not sure if this is the right sytax. please fix it if the syntax is wrong. thank you.

**Comment by matthew-ardi** [comment]
@sonminhtran1997 sometimes, serial read can read empty data due to the asynchronous nature of the communication. I think I can fix this from my side.

File: RealtimeAppBackend/RealtimeApp/views.py [hunk 2]
```diff
@@ -42,7 +43,56 @@ def retrieveData(request):
...
+        final_message = str(string.split('#', 1)[1])    # String of data transmitted through bluetooth from device
+    except:
+        return  # sometime exception might happen when server is trying to read serial data but no new data has been sent. 
+                # when this happen, skip the process below and let the server re-request serial data.
+
+    ##
+    ##  splitting messages to read boolean values of each event detection
+    ##
+    split_message = final_message.split('\n')
+    roomLightState = False
```
Comment (sonminhtran1997) [inline RealtimeAppBackend/RealtimeApp/views.py]: Why do you need to set everything to be false right here?

File: RealtimeAppBackend/RealtimeApp/views.py (hunk 1 above)
Comment (sonminhtran1997) [reply RealtimeAppBackend/RealtimeApp/views.py]: for this there is a function setData at the end of this file. Basically you just need to do this
setData(data, roomLightState, benchLightState, blenderState, spaceHeaterState, roomOccupiedState, grinderState)

Please help me update this function as you already added new fields

File: RealtimeAppBackend/RealtimeApp/views.py [hunk 3]
```diff
@@ -42,7 +43,56 @@ def retrieveData(request):
 @csrf_exempt
```
Comment (sonminhtran1997) [inline RealtimeAppBackend/RealtimeApp/views.py]: please help me remove this tag we do not need it anymore

File: RealtimeAppBackend/RealtimeApp/views.py [hunk 4]
```diff
@@ -42,7 +43,56 @@ def retrieveData(request):
...
 def retrieveBluetooth(request):
```
Comment (sonminhtran1997) [inline RealtimeAppBackend/RealtimeApp/views.py]: remove request argument too, we do not need it. Just leave it as a function and I will use this function later to get real time data

**Review by sonminhtran1997 — COMMENTED** [review]
> @sonminhtran1997 sometimes, serial read can read empty data due to the asynchronous nature of the communication. I think I can fix this from my side.

Got it, when you fix it just comment here so I can merge it. Also please fix those code review as I commented in the code. Just minor fix to match with my design in the front end.

**Comment by sonminhtran1997** [comment]
also after you finish please do pip freeze > requirements.txt to update dependencies.

**Pull request merged**